docs/source/functional.rst

```diff
@@ -39,8 +39,7 @@ An instance of :class:`.Functional`, ``f``, may provide three core operations.
    - ``f.grad(x)`` returns the gradient of the functional evaluated at ``x``.
    - Gradients are calculated using JAX reverse-mode automatic differentiation,
      exposed through :func:`scico.grad`.
-   - A functional that is smooth has the attribute ``f.is_smooth == True``.
-   - NOTE:  The gradient of a functional ``f`` can be evaluated even if ``f.is_smooth == False``.
+   - NOTE:  The gradient of a functional ``f`` can be evaluated even if that functional is not smooth.
```

Comment:
Can we refer to part of the jax docs to expand on this? Also, considering using markup (e.g. *Note*) rather than all caps for emphasis.

Reply:
I had a look this morning and could not find any clear discussion of this in the jax docs. I can at least confirm that the statement is sometimes true, e.g., `jax.grad(lambda x: jnp.abs(x))(0.0) == 1.0`.

Reply:
> the statement is sometimes true, e.g., `jax.grad(lambda x: jnp.abs(x))(0.0) == 1.0`

I think this is an easier example because x is defined everywhere else but at 0.

Where it breaks is e.g. at `x / (1 + jnp.sqrt(x))` which `grad`s to `nan` at zero. [See](https://jax.readthedocs.io/en/latest/notebooks/Custom_derivative_rules_for_Python_code.html?highlight=differentiable#enforcing-a-differentiation-convention)

I don't know what the general rule for scico is, though.

Reply:
> Also, considering using markup

See 0fdfcf6